engine/src/main/java/graphql/nadel/engine/execution/NadelExecutionStrategy.java

```diff
@@ -121,17 +121,44 @@ public NadelExecutionStrategy(List<Service> services,
         List<CompletableFuture<OneServiceExecution>> result = new ArrayList<>();
         for (MergedField mergedField : fieldSubSelection.getMergedSelectionSet().getSubFieldsList()) {
             ExecutionStepInfo fieldExecutionStepInfo = executionStepInfoFactory.newExecutionStepInfoForSubField(executionCtx, mergedField, rootExecutionStepInfo);
-            Service service = getServiceForFieldDefinition(fieldExecutionStepInfo.getFieldDefinition());
-
-            CreateServiceContextParams parameters = CreateServiceContextParams.newParameters()
-                    .from(executionCtx)
-                    .service(service)
-                    .executionStepInfo(fieldExecutionStepInfo)
-                    .build();
-
-            CompletableFuture<Object> serviceContextCF = serviceExecutionHooks.createServiceContext(parameters);
-            CompletableFuture<OneServiceExecution> serviceCF = serviceContextCF.thenApply(serviceContext -> new OneServiceExecution(service, serviceContext, fieldExecutionStepInfo));
-            result.add(serviceCF);
+            boolean isNamespaced = !fieldExecutionStepInfo.getFieldDefinition().getDirectives("namespaced").isEmpty();
+            if (isNamespaced) {
+                GraphQLObjectType namespacedObjectType = (GraphQLObjectType) fieldExecutionStepInfo.getUnwrappedNonNullType();
+                for (Map.Entry<Service, Set<GraphQLFieldDefinition>> serviceWithCorrespondingFieldDefinitions : fieldInfos.fieldDefinitionsByService.entrySet()) {
+                    Service service = serviceWithCorrespondingFieldDefinitions.getKey();
+                    Set<GraphQLFieldDefinition> secondLevelFieldDefinitionsForService = serviceWithCorrespondingFieldDefinitions.getValue();
+
+                    Optional<MergedField> newMergedField = MergedFieldUtil.includeSubSelection(mergedField, namespacedObjectType, executionCtx,
+                            field -> secondLevelFieldDefinitionsForService.stream()
+                                    .filter(namespacedObjectType.getFieldDefinitions()::contains)
+                                    .anyMatch(graphQLFieldDefinition -> graphQLFieldDefinition.getName().equals(field.getName())));
```

Comment:
I think we are still in precarious ground here.

Imagine a mutation `jira : JiraMutation` field that has a field called `issues` - this will match.

I really want us to do the lookup by `GraphqlObjectType` and field name so that we can narrow it down to a unique graphql field coordinate

This is where I think `FieldInfos` needs to help us

```
Map<Service,List<GraphFieldDefintiions> servicesAndFields =  fieldInfos.getByObjectType(namespacedObjectType)
```

We can then name match to our hearts content because we know its been narrowed down to ONLY the named fields of that object type

Reply:
as discussed, this does narrow it down to unique coordinates because of `.filter(namespacedObjectType.getFieldDefinitions()::contains)` on the line 133
I can extract it to `FieldInfos` anyway though